server/src/inmantals/server.py

```diff
@@ -54,16 +60,24 @@ def __init__(self, instream: BaseIOStream, outstream: BaseIOStream, address):
         self.threadpool = ThreadPoolExecutor(1)
         self.anchormap = None
         self.reverse_anchormap = None
+        self.types: Optional[Dict[str, inmanta_type.Type]] = None
         self.state_lock: asyncio.Lock = asyncio.Lock()
         self.diagnostics_cache: Optional[lsp_types.PublishDiagnosticsParams] = None
+        self.supported_symbol_kinds: Optional[Set[lsp_types.SymbolKind]] = None
 
-    async def initialize(self, rootPath, rootUri, **kwargs):  # noqa: N803
+    async def initialize(self, rootPath, rootUri, capabilities: Dict[str, object], **kwargs):  # noqa: N803
         logger.debug("Init: " + json.dumps(kwargs))
 
         self.rootPath = rootPath
         self.rootUrl = rootUri
         os.chdir(rootPath)
 
+        try:
+            value_set: List[int] = capabilities["workspace"]["symbol"]["symbolKind"]["valueSet"]  # type: ignore
+            self.supported_symbol_kinds = set(map(lsp_types.SymbolKind, value_set))
```

Comment:
To your first point: do you mean to load the defaults if none are explicitly passed by the client? I tried that at first but that makes it difficult to know whether it's explicit or implicit. Also, it turned out that I couldn't find a way to make it cleaner that way because I don't see how you could fully dynamic. So, I'm willing to change this but I would need a more concrete suggestion.
- Second point: will fix.

Reply:
I was not thinking about the scenario where the client doesn't provide any capabilities. I was more thinking about the scenario where the client provides a capability that the extension doesn't support. That would result in an exception right?

Reply:
That's right, I hadn't thought of that. I'll change this to ignore (and log a warning) symbol kinds that are not supported by the server.